datastore/src/main/kotlin/com/openlattice/collaborations/controllers/CollaborationController.kt

```diff
@@ -0,0 +1,189 @@
+import com.openlattice.collaborations.CollaborationsApi
+import com.openlattice.collaborations.CollaborationsApi.Companion.CONTROLLER
+import com.openlattice.collaborations.CollaborationsApi.Companion.DATABASE_PATH
+import com.openlattice.collaborations.CollaborationsApi.Companion.ID
+import com.openlattice.collaborations.CollaborationsApi.Companion.ID_PATH_PARAM
+import com.openlattice.collaborations.CollaborationsApi.Companion.ORGANIZATIONS_PATH
+import com.openlattice.collaborations.CollaborationsApi.Companion.ORGANIZATION_ID
+import com.openlattice.collaborations.CollaborationsApi.Companion.ORGANIZATION_ID_PATH_PARAM
+import com.openlattice.collaborations.CollaborationsApi.Companion.PROJECT_PATH
+import com.openlattice.collaborations.CollaborationsApi.Companion.TABLES_PATH
+import com.openlattice.collaborations.CollaborationsApi.Companion.TABLE_ID
+import com.openlattice.collaborations.CollaborationsApi.Companion.TABLE_ID_PATH_PARAM
+import com.openlattice.organizations.OrganizationDatabase
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.*
+import java.util.*
+import java.util.stream.Collectors
+import javax.inject.Inject
+
+@RestController
+@RequestMapping(CONTROLLER)
+@SuppressFBWarnings(
+        value = ["RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE"],
+        justification = "Allowing redundant kotlin null check on lateinit variables")
+class CollaborationController : AuthorizingComponent, CollaborationsApi {
+
+    @Inject
+    private lateinit var authorizations: AuthorizationManager
+
+    @Inject
+    private lateinit var collaborationService: CollaborationService
+
+    @Timed
+    @GetMapping(value = ["", "/"], produces = [MediaType.APPLICATION_JSON_VALUE])
+    override fun getCollaborations(): Iterable<Collaboration> {
+        val authorizedCollaborationIds = getAllAuthorizedCollaborationIds()
+        return collaborationService.getCollaborations(authorizedCollaborationIds).values
+    }
+
+    @Timed
+    @PostMapping(value = ["", "/"], consumes = [MediaType.APPLICATION_JSON_VALUE], produces = [MediaType.APPLICATION_JSON_VALUE])
+    override fun createCollaboration(@RequestBody collaboration: Collaboration): UUID {
+        return collaborationService.createCollaboration(collaboration, Principals.getCurrentUser())
+    }
+
+    @Timed
+    @GetMapping(value = [ID_PATH_PARAM], produces = [MediaType.APPLICATION_JSON_VALUE])
+    override fun getCollaboration(@PathVariable(ID) id: UUID): Collaboration {
+        ensureReadAccess(AclKey(id))
+        return collaborationService.getCollaboration(id)
+    }
+
+    @Timed
+    @GetMapping(value = [ORGANIZATIONS_PATH + ORGANIZATION_ID_PATH_PARAM], produces = [MediaType.APPLICATION_JSON_VALUE])
+    override fun getCollaborationsIncludingOrganization(@PathVariable(ORGANIZATION_ID) organizationId: UUID): Iterable<Collaboration> {
```

Comment:
The `Including` part is throwing me off. Is this the same thing as getting collaborations filtered by the given org id / getting collaborations that belong to an org? 

If yes, I think `getOrganizationCollaborations()` is more clear.

Reply:
Sort of, but the directionality model there is backwards. I also am not a huge fan of `includes` but I think "OrganizationCollaboration" is a bit misleading since it's really more of a membership model.